### PR TITLE
Update roadmap.yml

### PR DESCRIPTION
**Pull request by lucaswoj**
cc @mapsam @jfirebaugh @lbud @mollymerp @hjudge @alianthes

### CODE REVIEW

File: docs/_data/roadmap.yml [hunk 1]
```diff
@@ -1,33 +1,32 @@
...
   items:
     -
-      name: Improve performance around setting styles with <code>setStyle()</code>
-      issues: ["1370", "1989"]
+      name: Allow raster source "loading" pattern and "errored" pattern
+      issues: ["3280"]
     -
-      name: Fix GeoJSON memory leak
-      issues: ["3220", "3595"]
+      name: Add <code>circle-stroke-*</code> style properties
```
Comment (mapsam) [inline docs/_data/roadmap.yml]: Is there an issue to link to here?

**Comment by mapsam** [comment]
@lbud should #3580 be part of the Custom source type API project? https://github.com/mapbox/mapbox-gl-js/projects/2

File: docs/_data/roadmap.yml (hunk 1 above)
Comment (lucaswoj) [inline docs/_data/roadmap.yml]: Not yet 😬

**Comment by lucaswoj** [comment]
@mapsam The canvas layer type is a separate from the custom source types project.

**Comment by lucaswoj** [comment]
Just updated based on feedback 

 - add issue # for `circle-stroke-*`
 - add canvas layer type to "Under development"

**Pull request merged**